### DISCUSSION TITLE
Access to github using svn bridge and ssh key

### DISCUSSION THREAD

**Discussion by pryrt** [community/community#21619] [New to GitHub]
<p>I’ve always used the <a href="https://docs.github.com/en/free-pro-team@latest/github/importing-your-projects-to-github/support-for-subversion-clients" rel="noopener nofollow ugc">svn interface to GitHub</a> as much as possible, because my day job version control is all Subversion, so that’s the workflow I’m used to.</p>
<p>However, I just got the email saying that password access is going away, which will break the svn https checkout.</p>
<p>I added an ssh key to my GitHub account, and told putty where my key is.  I’ve run <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/testing-your-ssh-connection" rel="noopener nofollow ugc">the test</a> of using putty to prove that my key can access  GitHub without difficulty.</p>
<p>However, I haven’t figured out how to use the svn bridge over ssh for GitHub.  At work, when we access our work svn server with ssh key, I just use the <code>svn+ssh://...</code> protocol, but when I tried that for my GitHub repo, it wouldn’t connect.  The <code>SSH</code> tab on the code download button doesn’t give a valid svn-style URL.</p>
<p>I haven’t been able to find an official document for how to make the ssh key using the svn bridge, and my searches elsewhere haven’t found anything yet.</p>
<p>Is there an official or unofficial guide for how to get the ssh key working using the svn bridge to GitHub?</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>If it was working perfectly well using a password, you could try using a <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">personal access token</a>. This should work roughly the same way as a password (with https), but doesn’t grant full account access.</p>

**Comment by pryrt** [comment]
<p>Thanks, that was quite simple to set up and get working.</p>

**Comment by piotr-chowaniec** [comment]
<p>Hi There!<br>
Glad to know you’ve managed to solve <a class="mention" href="/pryrt">@pryrt</a> issue.<br>
However, it’d be worth knowing how to access svn with ssh. According to docs, it should be straightforward. copy and use ssh link which is available under the code dropdown button. However, as <a class="mention" href="/pryrt">@pryrt</a> mentioned in the original message, it does not work for me as well.<br>
Any thoughts?</p>
<p>Best regards!</p>

**Comment by canuckjacq** [comment]
<p>Hi and :wave: Welcome!</p>
<p>I’m afraid <a href="https://docs.github.com/en/github/importing-your-projects-to-github/support-for-subversion-clients">our docs are quite specific that we only support svn clients via HTTPS</a>:</p>
<blockquote>
<p>GitHub supports Subversion clients via the HTTPS protocol. We use a Subversion bridge to communicate svn commands to GitHub.</p>
</blockquote>

**Comment by piotr-chowaniec** [comment]
<p>What about <span class="hashtag">#3</span> ?</p>
<blockquote>
<p>To clone the repository using an SSH key, including a certificate issued by your organization’s SSH certificate authority, click <strong>Use SSH</strong>  …</p>
</blockquote>

**Comment by canuckjacq** [comment]
<p>Yup, that’s confusing. I’ve had a look at <a href="https://github.com/github/docs/edit/main/content/github/importing-your-projects-to-github/support-for-subversion-clients.md">the document</a>, which is open source (there’s a link to it at the bottom of the page).</p>
<p>It looks like this is a reusable snippet that is inserted everywhere that we tell folks how to clone a repository. In this case, it shouldn’t be used, because we have to use HTTPS with svn.</p>
<p>I’ll open an issue to get that fixed. Apologies for the confusion.</p>

**Comment by piotr-chowaniec** [comment]
<p>No worries. We’ll handle it another way then. Thanks for such a quick reply and clarifications :muscle: .</p>
